File: recovered/src/com/lumiyaviewer/lumiya/voice/common/messages/VoiceRinging.java
package com.lumiyaviewer.lumiya.voice.common.messages;

import android.net.Uri;
import android.os.Bundle;
import com.lumiyaviewer.lumiya.BuildConfig;
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage;
import com.lumiyaviewer.lumiya.voice.common.model.VoiceChannelInfo;
import java.util.UUID;
import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/* loaded from: classes.dex */
public class VoiceRinging implements VoicePluginMessage {

    @Nullable
    public final UUID agentUUID;

    @Nonnull
    public final String sessionHandle;
    public final VoiceChannelInfo voiceChannelInfo;

    public VoiceRinging(Uri uri) {
        this.sessionHandle = uri.getQueryParameter("sessionHandle");
        String queryParameter = uri.getQueryParameter("agentUUID");
        this.agentUUID = queryParameter != null ? UUID.fromString(queryParameter) : null;
        this.voiceChannelInfo = new VoiceChannelInfo(uri);
    }

    public VoiceRinging(Bundle bundle) {
        this.sessionHandle = bundle.getString("sessionHandle");
        this.voiceChannelInfo = new VoiceChannelInfo(bundle.getBundle("voiceChannelInfo"));
        String string = bundle.getString("agentUUID");
        this.agentUUID = string != null ? UUID.fromString(string) : null;
    }

    public VoiceRinging(@Nonnull String str, VoiceChannelInfo voiceChannelInfo, @Nullable UUID uuid) {
        this.sessionHandle = str;
        this.voiceChannelInfo = voiceChannelInfo;
        this.agentUUID = uuid;
    }

    @Override // com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage
    public Bundle toBundle() {
        Bundle bundle = new Bundle();
        bundle.putString("sessionHandle", this.sessionHandle);
        bundle.putBundle("voiceChannelInfo", this.voiceChannelInfo.toBundle());
        bundle.putString("agentUUID", this.agentUUID != null ? this.agentUUID.toString() : null);
        return bundle;
    }

    public Uri toUri() {
        Uri.Builder appendQueryParameter = new Uri.Builder().scheme(BuildConfig.APPLICATION_ID).authority("voice").appendQueryParameter("sessionHandle", this.sessionHandle);
        if (this.agentUUID != null) {
            appendQueryParameter.appendQueryParameter("agentUUID", this.agentUUID.toString());
        }
        this.voiceChannelInfo.appendToUri(appendQueryParameter);
        return appendQueryParameter.build();
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/voice/common/messages/VoiceRinging.kt
package com.lumiyaviewer.lumiya.voice.common.messages

import android.net.Uri
import android.os.Bundle
import com.lumiyaviewer.lumiya.BuildConfig
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage
import com.lumiyaviewer.lumiya.voice.common.model.VoiceChannelInfo
import java.util.UUID

class VoiceRinging : VoicePluginMessage {
    @JvmField
    val agentUUID: UUID?
    @JvmField
    val sessionHandle: String
    @JvmField
    val voiceChannelInfo: VoiceChannelInfo

    constructor(uri: Uri) {
        this.sessionHandle = uri.getQueryParameter("sessionHandle")!!
        val queryParameter = uri.getQueryParameter("agentUUID")
        this.agentUUID = if (queryParameter != null) UUID.fromString(queryParameter) else null
        this.voiceChannelInfo = VoiceChannelInfo(uri)
    }

    constructor(bundle: Bundle) {
        this.sessionHandle = bundle.getString("sessionHandle")!!
        this.voiceChannelInfo = VoiceChannelInfo(bundle.getBundle("voiceChannelInfo"))
        val string = bundle.getString("agentUUID")
        this.agentUUID = if (string != null) UUID.fromString(string) else null
    }

    constructor(sessionHandle: String, voiceChannelInfo: VoiceChannelInfo, uuid: UUID?) {
        this.sessionHandle = sessionHandle
        this.voiceChannelInfo = voiceChannelInfo
        this.agentUUID = uuid
    }

    override fun toBundle(): Bundle = Bundle().apply {
        putString("sessionHandle", sessionHandle)
        putBundle("voiceChannelInfo", voiceChannelInfo.toBundle())
        putString("agentUUID", agentUUID?.toString())
    }

    fun toUri(): Uri {
        val appendQueryParameter = Uri.Builder()
            .scheme(BuildConfig.APPLICATION_ID)
            .authority("voice")
            .appendQueryParameter("sessionHandle", sessionHandle)
        if (agentUUID != null) {
            appendQueryParameter.appendQueryParameter("agentUUID", agentUUID.toString())
        }
        voiceChannelInfo.appendToUri(appendQueryParameter)
        return appendQueryParameter.build()
    }
}
